src/main/java/com/k3ntako/HTTPServer/IOGenerator.java

```diff
@@ -0,0 +1,31 @@
+package com.k3ntako.HTTPServer;
+
+import com.k3ntako.HTTPServer.wrappers.PrintWriterWrapper;
+
+import java.io.*;
+import java.net.Socket;
+import java.util.HashMap;
+
+public class IOGenerator implements IOGeneratorInterface {
+
+  public HashMap<String, Object> generateIO(Socket clientSocket) {
```

Comment:
I think this is very close. A thought here is rather than returning the HashMap try returning a new IO Class(that has an interface we can mock). That class can have a couple of methods like read and write that are proxies for the actual calls to bufferedReader and PrintWriter.